### PR TITLE
PWN-7595 - First step of Bridge API integration

### PR DESCRIPTION
**Pull request by Davran331122**
add error wrapper and error mapper. Refactor JsonRpc request

## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7595

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/claim/api/response/BridgeBundleResponse.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    val erc20TokenAddress: EthAddress,
+    @SerializedName("expiresAt")
+    val expiresAt: String? = null,
+    @SerializedName("transactions")
+    val transactions: List<HexTransaction>,
+    @SerializedName("signatures")
+    val signatures: List<HexTransaction>? = null,
+    @SerializedName("fees")
+    val fees: BundleFeeResponse
+
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/api/response/BridgeBundleResponse.kt:25]:
```suggestion
```

File: app/src/main/java/org/p2p/wallet/claim/model/BridgeServiceError.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package org.p2p.wallet.claim.model
+
+sealed class BridgeServiceError {
+
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/model/BridgeServiceError.kt:4]:
```suggestion
```

can be sealed interface too

File: app/src/main/java/org/p2p/wallet/claim/model/HexTransaction.kt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package org.p2p.wallet.claim.model
+
+@JvmInline
+value class HexTransaction(val transactionInHexString: String)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/model/HexTransaction.kt:4]: i would rather extract it to :common / :core an make it just HexInstance / HexString for future reuse

File: app/src/main/java/org/p2p/wallet/claim/model/BridgeBundle.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import org.p2p.core.token.SolAddress
+import org.p2p.ethereumkit.internal.models.EthAddress
+
+data class BridgeBundle(
+    val bundleId: String,
+    val userWallet: EthAddress,
+    val recipient: SolAddress,
+    val token: EthAddress,
+    val expiresAt: String,
+    val transactions: List<String>,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/model/BridgeBundle.kt:12]: String  -> HexTransaction

File: ethereumkit/src/main/java/org/p2p/ethereumkit/internal/api/core/NodeWebSocket.kt [hunk 5]
```diff
@@ -115,88 +115,90 @@ class NodeWebSocket(
...
+                    }
+                    is Event.OnLifecycle -> {
+                        logger.info("On LifeCycle: $event")
+                    }
+                    else -> {
+                        logger.info("On Event: $event")
+                    }
+                }
+            }, { error ->
+                error.printStackTrace()
```
Comment (gslevinkov) [inline ethereumkit/src/main/java/org/p2p/ethereumkit/internal/api/core/NodeWebSocket.kt:166]: remove this please

File: ethereumkit/src/main/java/org/p2p/ethereumkit/internal/api/core/NodeWebSocket.kt [hunk 6]
```diff
@@ -115,88 +115,90 @@ class NodeWebSocket(
...
                         }
-                    } catch (error: Throwable) {
-                        logger.warning("Handle Response error: ${error.javaClass.simpleName}")
-                        error.printStackTrace()
                     }
-                }, { error ->
-                    logger.warning("On Response error: ${error.message ?: error.javaClass.simpleName}")
+                } catch (error: Throwable) {
+                    logger.warning("Handle Response error: ${error.javaClass.simpleName}")
                     error.printStackTrace()
```
Comment (gslevinkov) [inline ethereumkit/src/main/java/org/p2p/ethereumkit/internal/api/core/NodeWebSocket.kt:190]: remove this please

Comment (gslevinkov) [reply ethereumkit/src/main/java/org/p2p/ethereumkit/internal/api/core/NodeWebSocket.kt:190]: printStackTrace should be removed everywhere

File: ethereumkit/src/main/java/org/p2p/ethereumkit/internal/network/GsonTypeAdapters.kt [hunk 7]
```diff
@@ -114,6 +115,29 @@ class AddressTypeAdapter : TypeAdapter<EthAddress?>() {
     }
 }
 
+class SolAddressTypeAdapter : TypeAdapter<SolAddress?>() {
```
Comment (gslevinkov) [inline ethereumkit/src/main/java/org/p2p/ethereumkit/internal/network/GsonTypeAdapters.kt:118]: internal?

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**